server/application/application.go

```diff
@@ -717,16 +705,16 @@ func (s *Server) Delete(ctx context.Context, q *application.ApplicationDeleteReq
 		return nil, err
 	}
 
-	if q.Cascade != nil && !*q.Cascade && q.GetPropagationPolicy() != "" {
+	if !q.Cascade && q.GetPropagationPolicy() != "" {
```

Comment:
I'm concerned about the 2->3 proto upgrade because we've lost the ability to detect when boolean fields were omitted as opposed to defaulted false. Argo CD has logic that treats omitted fields differently than false, such as the original code here. I believe this is changing the behavior of cascaded deletion here.

Reply:
Please see more information about this design decision in proto3:
https://github.com/protocolbuffers/protobuf/issues/359

Reply:
OK, so @alexmt just informed me that starting with proto 3.15, fields can once again be marked as `optional`. I think what needs to happen in this PR is to specify `optional` to fields like Cascade, so that we get back the original behavior. I believe this will generate `Has()` methods to indicate if the field was explicitly set or not.

See: https://stackoverflow.com/a/62566052